extensions/markdown-language-features/src/test/markdownLink.test.ts

```diff
@@ -0,0 +1,49 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as assert from 'assert';
+import * as vscode from 'vscode';
+import { checkSmartPaste, createLinkSnippet } from '../languageFeatures/copyFiles/shared';
+import 'mocha';
+
+suite('getMarkdownLink', () => {
+
+	// implement SkinnyTextDocument
+	// end to end test of checkSmartPaste & createLinkSnippet
+
+	// check paste with selection
+	// check paste over markdown link
+	// check paste over markdown image
+	// check paste within markdown link
+	// check paste with open bracket
+
+	// check multicursor
+	// check image link
+	// check each case of smart paste
+
+
+	// check smart paste
+	test('Check smart paste with plain text', async () => {
```

Comment:
With these test name, try to describe what the test is actually doing too. For example `start parse should be enabled for plain text`